core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -524,15 +757,16 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
             producerIdCopy = txnMetadata.producerId
             producerEpochCopy = txnMetadata.producerEpoch
             // PrepareEpochFence has slightly different epoch bumping logic so don't include it here.
-            val currentTxnMetadataIsAtLeastTransactionsV2 = !txnMetadata.pendingState.contains(PrepareEpochFence) && txnMetadata.clientTransactionVersion.supportsEpochBump()
+            // Note that, it can only happen when the current state is Ongoing.
+            val isPrepareEpochFence = txnMetadata.pendingState.contains(PrepareEpochFence)
```

Comment:
Yeah, you are right, there are the same. When the pending state is PrepareEpochFence, it will only go to the `Ongoing` handling where we set the `isEpochFence` to true.